// pages/createArticle/index.js
import React, {useState, useEffect, useContext} from 'pages/createArticle/node_modules/react'
import {Redirect} from 'pages/createArticle/node_modules/react-router-dom'

import ArticleForm from 'pages/createArticle/node_modules/components/articleForm'
import useFetch from 'pages/createArticle/node_modules/hooks/useFetch'
import {CurrentUserContext} from 'pages/createArticle/node_modules/contexts/currentUser'

const CreateArticle = () => {
  const apiUrl = '/articles'
  const [isSuccessfullSubmit, setIsSuccessfullSubmit] = useState(false)
  const [{response, error}, doFetch] = useFetch(apiUrl)
  const [currentUserState] = useContext(CurrentUserContext)

  const onSubmit = article => {
    doFetch({
      method: 'post',
      data: {
        article
      }
    })
  }
  const initialValues = {
    title: '',
    description: '',
    body: '',
    tagList: ''
  }

  useEffect(() => {
    if (!response) {
      return
    }
    setIsSuccessfullSubmit(true)
  }, [response])

  if (currentUserState.isLoggedIn === null) {
    return null
  }

  if (isSuccessfullSubmit || currentUserState.isLoggedIn === false) {
    return <Redirect to="/" />
  }

  return (
    <div>
      <ArticleForm
        onSubmit={onSubmit}
        initialValues={initialValues}
        errors={(error && error.errors) || {}}
      />
    </div>
  )
}

export default CreateArticle
